### PR TITLE
Mint custom datapublications #1529 #1531

### LINKED ISSUE #1529: User can click a button to mint a DOI for their selection
**Description**:
Show a mint DOI button on the cart page, so that when a user clicks on the button it starts the DOI minting process. The button should only be enabled if the user has the permission to mint every item in their cart.

**Acceptance criteria**:
- [ ] A mint button is on the cart page
- [ ] When clicked, the user goes to the next step in the DOI minting process for that cart
- [ ] Button is disabled if the user doesn't have permission to mint an item in their cart

### LINKED ISSUE #1531: Add Generate DOI form
**Description**:
Add a form that is accessed when a user clicks the generate DOI button from the cart (see #1529). This form should be preceeded by a page asking the user to accept the data policy and they understand what they're doing by minting a DOI and they take responsibility for it. The form itself should have the following fields for users to edit:

- DOI title
- DOI description
- Creators
   - Creators should autopopulate with users corresponding to the users on the visits in the selection
   - User can remove any users but can't remove themselves (they're autopopulated as the PI)
   - Each creator has a first name, a surname, an affiliation and an email
   - Creators can be "Personal" or "Organisation"s

We must ensure all these fields are filled out before allowing the user to click the generate DOI button.

We should also show a summary of the items in the selection to the side of the form

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
Adds a button to the cart page which lets the user mint their selection if they have permission, and once clicked it takes them to an "accept data policy page", which then accepting takes them to the generate DOI form. Once the user has filled out the required fields, they can generate a DOI, which once it is minted will cause a popup to appear with the DOI and a button to take the user to the landing page. See screenshots of the generate DOI page (names and emails redacted in the screenshots since it's based on real users)

generate_doi_page

generate_doi_page_confirmation

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] Deploy with other PR & Alex's minting backend to test the DOI minting workflow

## Agile board tracking
Closes #1529, Closes #1531

### CODE REVIEW

**Comment by louise-davies** [comment]
I still need to do e2e tests, but I need to add the DOI minter API to the CI workflow for that

**Comment by louise-davies** [comment]
TODO: add contributors to the users list & add the related DOIs section

image

image

**Comment by louise-davies** [comment]
@patrick-austin I have added the functionality for Contributors & Related DOIs based off our previous conversation & the wireframes I mocked up:

image


image

**Comment by louise-davies** [comment]
@kaperoo For testing, you will need to add the following new urls to the download settings json

```
"doiMinterUrl": "https://scigateway-preprod.esc.rl.ac.uk/doi-minter",
"dataCiteUrl": "https://api.test.datacite.org",
```

File: packages/datagateway-dataview/cypress/e2e/card/dls/datasets.cy.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ describe('DLS - Datasets Cards', () => {
     cy.get('[data-testid="card"]').first().contains('DATASET 61');
   });
 
-  it('should be able to filter by multiple fields', () => {
+  it.only('should be able to filter by multiple fields', () => {
```
Comment (kaperoo) [inline packages/datagateway-dataview/cypress/e2e/card/dls/datasets.cy.ts:112]: remove `.only`

**Review by kaperoo — COMMENTED** [review]
Looks good but I think `settings` need to be defined in the _DownloadCartTable_ component to fix testing errors.

File: packages/datagateway-download/src/downloadCart/downloadCartTable.component.tsx
[the remark this answers is not in the document]
Comment (louise-davies) [reply packages/datagateway-download/src/downloadCart/downloadCartTable.component.tsx:1]: Yep, I think it got obliterated in the merge resolution

**Review by kaperoo — APPROVED** [review]

**Pull request merged**